File: java/IEC61968/Assets/ProcedureDataSet.java
package IEC61968.Assets;

import IEC61970.Base.Domain.DateTime;
import IEC61970.Base.Meas.MeasurementValue;
import IEC61968.InfIEC61968.InfAssets.TransformerObservation;
import IEC61968.Work.WorkTask;
import IEC61968.Common.Document;

/**
 * A data set recorded each time a procedure is executed. Observed results are
 * captured in associated measurement values and/or values for properties relevant
 * to the type of procedure performed.
 * @created 03-Jan-2024 12:06:14 PM
 */
public class ProcedureDataSet extends Document {

	/**
	 * Date and time procedure was completed.
	 */
	public DateTime completedDateTime;
	/**
	 * Asset to which this procedure data set applies.
	 */
	public Asset Asset;
	public MeasurementValue MeasurementValue;
	public TransformerObservation TransformerObservations;
	/**
	 * Work task that created this procedure data set.
	 */
	public WorkTask WorkTask;

	public ProcedureDataSet(){

	}
}//end ProcedureDataSet

File: py/IEC61968/Assets/ProcedureDataSet.py
from IEC61968.Assets.Asset import Asset
from IEC61968.Common.Document import Document
from IEC61968.InfIEC61968.InfAssets.TransformerObservation import TransformerObservation
from IEC61968.Work.WorkTask import WorkTask
from IEC61970.Base.Domain.DateTime import DateTime
from IEC61970.Base.Meas.MeasurementValue import MeasurementValue


class ProcedureDataSet(Document):
    """
    A data set recorded each time a procedure is executed. Observed results are
    captured in associated measurement values and/or values for properties relevant
    to the type of procedure performed.
    """

    def __init__(self):
        """
        Constructor for ProcedureDataSet.
        """
        super().__init__()
        self.completed_date_time = DateTime()
        """
        Date and time procedure was completed.
        """

        self.asset = Asset()
        """
        Asset to which this procedure data set applies.
        """

        self.measurement_value = MeasurementValue()
        """
        Measurement value associated with this procedure data set.
        """

        self.transformer_observations = TransformerObservation()
        """
        Transformer observations associated with this procedure data set.
        """

        self.work_task = WorkTask()
        """
        Work task that created this procedure data set.
        """
